src/discounts/components/DiscountRules/componenets/RuleForm/components/RuleConditions/hooks/useProductSearch.ts

```diff
@@ -0,0 +1,35 @@
+import { DEFAULT_INITIAL_SEARCH_DATA } from "@dashboard/config";
+import { CommonSearchOpts } from "@dashboard/hooks/makeTopLevelSearch/types";
+import { getSearchFetchMoreProps } from "@dashboard/hooks/makeTopLevelSearch/utils";
+import useProductSearchQuery from "@dashboard/searches/useProductSearch";
+import { mapEdgesToItems } from "@dashboard/utils/maps";
+
+export const useProductSearch = (channel: string) => {
```

Comment:
**issue**: I believe this should be named differently, now it's overlapping with top-level search `@dashboard/searches/useProductSearch` and might be confusing